WT-10729 Support a remove option for WT utility write command

lukech opened the pull request:
Add support of a new "-r" option for the WT utility "write" command, to remove an existing record. 

`$ ./wt -h <data_folder> write -r <uri> <existing_key>`

lukech commented:
evergreen retry

etienneptl commented on src/docs/command-line.dox:
```diff
@@ -512,25 +512,30 @@ The \c write command stores records into the specified data source.
+If the \c write command is called with the \c -a option, each command-line
+argument is a single value to be appended to the specified column-store
+data source. If the \c write command is called with the \c -r option,
+the command-line argument is an existing key. If the \c write command is
```
Does it need to be an existing key with `-r` option?

lukech replied:
Actually one can provide any key, and only an existing key is meaningful to remove a record, a non-existing key would lead to a WT_NOTFOUND error. I can update the text by removing the "existing" word.

etienneptl replied:
Yes, please I would like that. The user is free to give any key.

lukech replied:
Updated in [d8c1c28](https://github.com/wiredtiger/wiredtiger/pull/8906/commits/d8c1c28361791483d2c666019b3b37a180c0e942)

etienneptl reviewed (commented):
Nice one, could you update test_util12.py with some testing?

marcbutler commented on src/utilities/util_write.c:
```diff
@@ -16,10 +16,10 @@ static int
 usage(void)
```
I guess there is nowhere else to place this, but using *write* to remove is not at all intuitive.
I'd suggest a separate command, to help avoid accidental misuse, but I don't have a sense as to whether the additional work required is merited.

lukech replied:
I thought about that previously as well. The _write_ command actually has already supported "adding new records" before this PR implicitly (if the key provided is new). That gives me a reason to add the "removing an existing record" to the _write_ command as a command option. Does it make sense to treat _write_ as a general term like _update_?

marcbutler replied:
I think it is grey. I don't think "adding a new record" is a good analog, but I do see "allow overwrite" which is destructive, and remove might be thought of as "overwrite with nothing". To me it's a stretch and not intuitive, but given the additional work that would be needed to add an entire command this seems sensible.

lukech commented:
> Nice one, could you update test_util12.py with some testing?

@etienneptl good suggestion, updated the test in [423d3cf](https://github.com/wiredtiger/wiredtiger/pull/8906/commits/423d3cf999c751362570b7668c24d95cd3715a51). Also fixed a class name typo in test_util22.py.

lukech commented on src/utilities/util_write.c:
```diff
@@ -57,14 +60,22 @@ util_write(WT_SESSION *session, int argc, char *argv[])
             return (usage());
-    } else if (argc < 3 || ((argc - 1) % 2 != 0))
-        return (usage());
+    } else if (remove) {
```
Thanks for the suggestion - that's actually the change I made initially. Then I realized `append` expects **more than** 2 args (a list of values), while `remove` expects **exactly** 2 args (URI followed by a single key).

etienneptl reviewed (approved):
Can we add a comment here saying we expect the command to fail as the `-o` option is not given and the key `def` already exists?

etienneptl reviewed (approved):
LGTM, thanks for improving the doc as well.

lukech commented:
> Can we add a comment here saying we expect the command to fail as the -o option is not given and the key def already exists?

Added a few comments in [72f5566](https://github.com/wiredtiger/wiredtiger/pull/8906/commits/72f556634ce0e9d303066a40391a646afc4c201c)

marcbutler reviewed (commented):
lgtm
